src/App.js

```diff
@@ -1,22 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
+import { api } from './api'
 import { Container, Grid, Typography, IconButton } from '@mui/material';
 import AddIcon from '@mui/icons-material/Add';
 
 import AvatarHeader from './components/AvatarHeader';
 import NavBar from './components/NavBar';
 import StatsCard from './components/StatsCard';
 import CategoryCard from './components/CategoryCard';
+import CategoryModal from './components/CategoryModal';
 
 function App() {
-  const handleOnCategoryClick = () => {alert("Tıklandı")}
+  const [categories, setCategories] = useState([]);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const handleOpenModal = () => {
+    setModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+
+  useEffect(() => {
+    api().get('/category/all')
+      .then((response) => {
+        setCategories(...categories, response.data.data)
+      })
+      .catch((error) => {
+        alert(`Error: ${error.response.data.error.message}`)
+      });
+  }, [])
+
+  const handleAddCategory = (values) => {
+    api().post('/category', {
+      title: values.category,
+      emoji: 'home'
+    })
+      .then((response) => {
+        setCategories([...categories, { id: response.data.data.id, emoji: 'home', title: values.category, count: 0 }]);
```

Comment:
Böyle bir yöntem yerine tekrar backend servisinden /category/all ile yapsan daha doğru olur. Çünkü source of truth DB yani backend.